src/libraries/System.Private.CoreLib/src/System/DateTime.Windows.cs

```diff
@@ -15,150 +15,127 @@ public static unsafe DateTime UtcNow
         {
             get
             {
-                long fileTime;
+                ulong fileTime;
                 s_pfnGetSystemTimeAsFileTime(&fileTime);
 
                 if (s_systemSupportsLeapSeconds)
                 {
-                    FullSystemTime time;
+                    Interop.Kernel32.SYSTEMTIME time;
+                    ulong hundredNanoSecond;
 
-                    if (Interop.Kernel32.FileTimeToSystemTime(&fileTime, &time.systemTime) != Interop.BOOL.FALSE)
+                    if (Interop.Kernel32.FileTimeToSystemTime(&fileTime, &time) != Interop.BOOL.FALSE)
                     {
                         // to keep the time precision
-                        time.hundredNanoSecond = fileTime % 10000; // 10000 is the number of 100-nano seconds per Millisecond
+                        ulong tmp = fileTime;
```

Comment:
>tmp [](start = 30, length = 3)

what is the benefit having this tmp?

Reply:
This avoids a duplicate memory access for the remainder calculation.

Reply:
Cool. do you mind add a little comment on that to help the code readers understand it?